rpc/block.go

```diff
@@ -155,9 +155,10 @@ type BlockMetadata struct {
 
 // GetBlock returns information about a Tezos block
 // https://tezos.gitlab.io/mainnet/api/rpc.html#get-block-id
-func (c *Client) GetBlock(ctx context.Context, blockID tezos.BlockHash) (*Block, error) {
+func (c *Client) GetBlock(ctx context.Context, blockID interface{}) (*Block, error) {
 	var block Block
-	u := fmt.Sprintf("chains/%s/blocks/%s", c.ChainID, blockID)
+
+	u := fmt.Sprintf("chains/%s/blocks/%s", c.ChainID, GetBlockIDAsString(blockID))
```

Comment:
`GetBlockIDAsString()` can return an error message string which would directly be appended to the URL and sent out as HTTP request.

A better pattern is to define it as `func GetBlockIDAsString() (string, error) {}` and check for error condition before using the string in a URL.